### PR TITLE
Reduce overhead of upgrades for users with custom stylesheets/JS

### PR DESCRIPTION
**Pull request by techknowlogick**
I have custom JS/CSS in use on my gitea instance and when I upgrade I have to edit head.tmpl and footer.tmpl to add in my changes again (otherwise I might have out of date templates). This will allow me to put all of my changes in header_extras.tmpl/footer_extras.tmpl and not worry about upgrading.

Open to suggestions regarding the names of the additional templates, or their locations.

### CODE REVIEW

File: templates/base/footer.tmpl [hunk 1]
```diff
@@ -66,5 +66,6 @@
 	<!-- JavaScript -->
 	<script src="{{AppSubUrl}}/vendor/plugins/semantic/semantic.min.js"></script>
 	<script src="{{AppSubUrl}}/js/index.js?v={{MD5 AppVer}}"></script>
+{{template "base/footer_extras" .}}
```
Comment (sapk) [inline templates/base/footer.tmpl]: Cleaner to create a folder extra at root of templates containing footer.tmpl and header.tmpl.

**Review by sapk — CHANGES_REQUESTED** [review]
A little change needed and add 2-3 lines (in https://github.com/go-gitea/gitea/blob/master/docs/content/doc/advanced/customizing-gitea.en-us.md) recommending to use extra folder to add some custom header and footer resources to keep template updated.

**Comment by techknowlogick** [comment]
@sapk thanks for the feedback. I've changed the directory for the extras template files, and added in some documentation on when to use them.

**Review by sapk — APPROVED** [review]

**Comment by sapk** [comment]
LGTM

**Comment by ethantkoenig** [comment]
Would `custom` be a better directory name than `extras`?

**Comment by techknowlogick** [comment]
@ethantkoenig Thanks for feedback. As `custom` is already a top level dir, when we tell people to go to the custom folder it might create confusion to have multiple dirs with the same name. (See use of `custom` on this page: https://docs.gitea.io/en-us/customizing-gitea/ )

**Comment by ethantkoenig** [comment]
@techknowlogick Could we explicitly tell them to go to `templates/custom`, so they don't get confused?

**Comment by techknowlogick** [comment]
@ethantkoenig yes, but then for the other custom directory we have to explicitly say not the `templates/custom` directory. For example we have instructions for adding the robots.txt, and serving public files to put all of those in `custom` or `custom/public`. There is also a user on discord that is currently confused about just even the one custom directory.

However, this is as much commitment to the idea of that folder name as I have, and so in light of the above if you still want me to change the directory name I will.

**Comment by ethantkoenig** [comment]
I personally prefer `custom` over `extras`, for two reasons:
1. I think it more clearly describes the intent of the directory (customized template changes). If I saw the name `extras`, my first instinct would be that it is a directory of miscellaneous templates, not customizable ones.
2. For consistency with the existing `custom` directory; every directory named `custom` is for some sort of custom change.

**Comment by techknowlogick** [comment]
Ok. Sounds good. Im out right now, but will change it when I get home.

**Comment by techknowlogick** [comment]
@ethantkoenig I've updated it. Thanks again for the feedback.

cc: @lafriks

**Comment by lafriks** [comment]
LGTM

**Pull request merged**